torch/_inductor/codegen/cpp_utils.py

```diff
@@ -492,9 +494,9 @@ def localize_nodes(
         The the data access of `local_buf` is assumed to be contiguous with the
         same order as the `global_buf`.
         """
-        local_buffers = list(self.local_buffers.values())
-        global_buffers = list(self.local_to_global.values())
-        assert len(global_buffers[0].get_size()) == len(local_buffers[0].get_size())
+        for global_buffer_name, global_buffer in self.global_buffers.items():
+            local_buffer = self.global_to_local[global_buffer_name]
+            assert len(global_buffer.get_size()) == len(local_buffer.get_size())
```

Comment:
Perhaps we don't need this strict check on size equivalence? Local buffer can be smaller than the global one?

Reply:
Sure, remove this check.